### PR TITLE
Add Dilated BinaryDenseNet37

### PR DESCRIPTION
**Pull request by lgeiger**
This adds a dilated version of BinaryDenseNet37 which achieves ~64.3 % top-1 accuracy.

### CODE REVIEW

File: larq_zoo/densenet.py [hunk 1]
```diff
@@ -49,11 +50,12 @@ def binary_densenet(
 
     for block, layers_per_block in enumerate(args.layers):
         for _ in range(layers_per_block):
-            x = densely_connected_block(x, args)
+            x = densely_connected_block(x, args, args.dilation_rate[block])
 
         if block < len(args.layers) - 1:
             x = keras.layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
-            x = keras.layers.MaxPooling2D(2, strides=2)(x)
+            if args.dilation_rate[block + 1] == 1:
```
Comment (timdebruin) [inline larq_zoo/densenet.py]: Should we consider the case where some but not all of the max pooling layers are replaced with dilation? (e.g. dilation_rate = [1, 1, 2, 2])

Comment (lgeiger) [reply larq_zoo/densenet.py]: I tried to keep this close to the [official implementation](https://github.com/hpi-xnor/BMXNet-v2-examples/blob/master/binary_models/densenet.py#L84-L87).

I think for now it is fine to only use dilation when we don't reduce the spacial dimension in the transition block.

File: tests/train_test.py [hunk 2]
```diff
@@ -1,7 +1,13 @@
+import pytest
 from click.testing import CliRunner
 from larq_zoo import train
+import tensorflow as tf
 
 
+@pytest.mark.skipif(
+    int(tf.__version__[0]) == 2,
+    reason="This currently fails on TF 2 due to https://github.com/larq/larq/pull/195.",
+)
```
Comment (timdebruin) [inline tests/train_test.py]: Should we test for tf version when making the model for which tf2 fails and raise a comprehensible error preemptively?

Comment (lgeiger) [reply tests/train_test.py]: I don't think that's necessary for now, since TF 2 isn't stable yet and this might be a bug in TensorFlow itself. Only the training code fails, so loading the model via the public API works correctly in this version.

We can add a proper error message once we have time to dig into https://github.com/larq/larq/pull/195
